packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -0,0 +1,147 @@
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/utils/SafeCast.sol";
+
+import "./interfaces/IFederatedAttestations.sol";
+import "../common/interfaces/IAccounts.sol";
+import "../common/interfaces/ICeloVersionedContract.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "../common/Signatures.sol";
+import "../common/UsingPrecompiles.sol";
+import "../common/libraries/ReentrancyGuard.sol";
+
+/**
+ * @title Contract mapping identifiers to accounts
+ */
+contract FederatedAttestations is
+  IFederatedAttestations,
+  ICeloVersionedContract,
+  Ownable,
+  Initializable,
+  UsingRegistry,
+  ReentrancyGuard,
+  UsingPrecompiles
+{
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+
+  struct Attestation {
+    address account;
+    uint256 issuedOn;
+    address signer;
+  }
+  // identifier -> issuer -> attestations
+  mapping(bytes32 => mapping(address => Attestation[])) public identifierToAddresses;
+  // account -> issuer -> identifiers
+  mapping(address => mapping(address => bytes32[])) public addressToIdentifiers;
+  // signer => revocation time
+  mapping(address => uint256) public revokedSigners;
```

Comment:
can you add a method signature for revokingSigners? logic doesn't have to be implemented yet but would make it a bit easier/cleaner to get started on the CK wrappers IMO, lmk what you think!

Reply:
do we also want to be able to un-revoke signers? is revoking permanent?

Reply:
hmm I think revoking is likely permanent, since the keys would be compromised? Personally think un-revoking is probably for few enough use cases that issuers should just register a new signer